controllers/credential_controller_test.go

```diff
@@ -0,0 +1,274 @@
+package controllers
+
+import (
+	"context"
+	"time"
+
+	civ1 "github.com/estafette/estafette-ci-operator/api/v1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	yaml "gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("Credential Controller", func() {
+
+	const timeout = time.Second * 10
+	const interval = time.Second * 1
+	const charset = "abcdefghijklmnopqrstuvwxyz"
+
+	var credentialKeyName = "credential-" + randomStringWithCharset(10, charset)
+	var privateCredentialKeyName = "credential-private-" + randomStringWithCharset(10, charset)
+	var gkeCredentialKeyName = "credential-gke-" + randomStringWithCharset(10, charset)
+
+	BeforeEach(func() {
+		// failed test runs that don't clean up leave resources behind.
+		keys := []string{credentialKeyName, privateCredentialKeyName}
+		for _, value := range keys {
+
+			cred := &civ1.Credential{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      value,
+					Namespace: "default",
+				},
+			}
+
+			_ = k8sClient.Delete(context.Background(), cred)
+		}
+	})
+
+	AfterEach(func() {
+		// Add any teardown steps that needs to be executed after each test
+		keys := []string{credentialKeyName, privateCredentialKeyName}
+		for _, value := range keys {
+
+			cred := &civ1.Credential{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      value,
+					Namespace: "default",
+				},
+			}
+
+			_ = k8sClient.Delete(context.Background(), cred)
+		}
+	})
+
+	Context("One crededential", func() {
+		It("Should handle credential correctly", func() {
```

Comment:
What is handling it correctly? Perhaps be a bit more explicit in saying what is correct behaviour.